audio/out/ao_wasapi.c

```diff
@@ -219,16 +227,23 @@ static void uninit(struct ao *ao)
     MP_DBG(ao, "Uninit wasapi\n");
     struct wasapi_state *state = ao->priv;
     wasapi_release_proxies(state);
-    if (state->hUninit)
-        SetEvent(state->hUninit);
+    if (state->hWake)
+        set_thread_state(ao, WASAPI_THREAD_SHUTDOWN);
+
     /* wait up to 10 seconds */
-    if (WaitForSingleObject(state->threadLoop, 10000) == WAIT_TIMEOUT) {
+    if (state->hAudioThread &&
+        WaitForSingleObject(state->hAudioThread, 10000) == WAIT_TIMEOUT)
+    {
```

Comment:
Haven't seen this code before, but using a timeout kind of looks like a bad idea. IMO it should just wait infinitely, because the timeout case can't be handled and would just result in a crash (or fun memory corruption) later on. Probably out of scope of this PR, though.